capitolwords_ng/parsers/crec.py

```diff
@@ -111,24 +105,28 @@ def parse_mods_file(self, mods_file):
         for record in records:
             if (record['ID'].split('-')[-1].startswith('PgD') or
                 record['ID'].split('-')[-2].startswith('PgD') or
+                record['ID'].split('-')[-1].startswith('FrontMatter') or
                 record['content'] is None or
                 'content' not in record.keys()):
-                # dont process daily digests
+                # dont process daily digests or front matters or pages with no content
                 continue
 
             text = preprocess(record['content'])
             textacy_text = textacy.Doc(self.nlp(text))
+            
+            # Extract named entities and their types & frequencies
             named_entities = get_named_entities(textacy_text)
+            record['named_entities'] = ''
             if any(named_entities):
                 named_entity_types = get_named_entity_types(named_entities)
                 named_entity_freqs = get_named_entity_frequencies(named_entities)
-                # for ne in sorted(named_entity_freqs, key=named_entity_freqs.get, reverse=True):
-                #     record['|'.join([ne, named_entity_types[ne]])] = named_entity_freqs[ne]
                 record['named_entities'] = str([
```

Comment:
With the new conditions this comprehension is getting a little unwieldy, probably time to break it out into a proper for loop.